code/object/objectshield.cpp

```diff
@@ -17,34 +17,69 @@
 
 #include <limits.h>
 
+// Private variables
+static const float shield_scale_factor = static_cast<float>(1.0 / (log(50.0) - log(1.0)));	// Factor used in Goober5000's scale_quad
 
-float shield_get_strength(object *objp)
-{
-	// no shield system, no strength!
-	if (objp->flags[Object::Object_Flags::No_shields])
+// Private Function Declarations
+/**
+ * @brief Logarithmically scales a quadrant's strength according to the shield generator's health %
+ *
+ * @param[in] generator_fraction The shield generator's HP in %
+ * @param[in] quad_strength      The quadrant's strength, before scaling
+ *
+ * @returns The quadrant's effective strength.
+ *
+ * @author Goober5000
+ */
+float scale_quad(float generator_fraction, float quad_strength);
```

Comment:
Why is this function declared here and the immediately defined a few lines after this? The declaration and definition could be merged here.